chain-abci/src/staking/table.rs

```diff
@@ -170,7 +170,13 @@ impl StakingTable {
         block_height: BlockHeight,
         voters: &[(TendermintValidatorAddress, bool)],
         evidences: &[(TendermintValidatorAddress, BlockHeight, Timespec)],
-    ) -> Vec<(StakedStateAddress, Coin, PunishmentKind)> {
+    ) -> Vec<(
+        StakedStateAddress,
+        Coin,
+        Coin,
+        PunishmentKind,
+        Option<Timespec>,
```

Comment:
It'd be good to have some type alias for this -- also what's difference between the `Coin` on the second position and the third position

Reply:
Created an alias `PunishmentKind` and refactored the `(Coin, Coin)` pair to `SlashedCoin` structure for readability.